compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignTiles.cpp

```diff
@@ -433,8 +448,67 @@ void AMDAIEAssignTilesPass::runOnOperation() {
   LLVM_DEBUG(llvm::dbgs() << "After duplicateGlobalObjFifos: \n"
                           << *parentOp << "\n");
 
+  // Analyse the count of each L3 logicalObjectFifos appearing in L3<->L2
+  // CopyOp. We would be maintaining this count in a map `l3BufferCount` and
+  // using this later to pick one tile to assign to the L3 logicalObjectFifo.
+  // TODO(avarma): Have an analysis pass instead and annotate the L3 buffers
+  // with their count. OR better to have an analysis pass that annotates
+  // LHS/RHS/OUT buffers. That might come handy for other passes.
+  bool visitedCopyOp = false;
+  DenseMap<Operation *, int64_t> l3BufferCount;
+  auto res = parentOp->walk([&](Operation *op) -> WalkResult {
+    if (auto copyOp = dyn_cast<CopyOpInterface>(op)) {
+      visitedCopyOp = true;
+      auto source = dyn_cast_if_present<AMDAIE::LogicalObjFifoOpInterface>(
+          copyOp.getSource().getDefiningOp());
+      if (!source) {
+        visitedCopyOp = false;
+        return WalkResult::interrupt();
+      }
+      uint8_t memSpace = source.getMemorySpaceAsUInt();
+      auto fromMemrefOp = dyn_cast<AMDAIE::LogicalObjectFifoFromMemrefOp>(
+          source.getOperation());
+      if (memSpace == 0 && fromMemrefOp) {
+        Operation *defOp = fromMemrefOp.getMemref().getDefiningOp();
+        if (!l3BufferCount.contains(defOp)) l3BufferCount[defOp] = 0;
+        l3BufferCount[defOp] += 1;
+      }
+      return WalkResult::advance();
+    } else if (isa<AMDAIE::CoreOp>(op)) {
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+
+  // We don't need the following. Have added this just for sake of completeness
+  // in the entire L3 buffer count analysis.
```

Comment:
Remove this note as it's bound to get stale at some point when we silently need it. It's good to always add logic for completeness and a test case for it as well, even though we haven't seen the case in an e2e example.